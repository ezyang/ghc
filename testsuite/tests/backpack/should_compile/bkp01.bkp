package p-impls where
    module P(hello) where
        hello = "Hello "
    module Q(hello, world) where
        import P
        world = "World"

package q where
    include p-impls
    module Main where
        import P
        import Q
        main = putStrLn (hello ++ world)
