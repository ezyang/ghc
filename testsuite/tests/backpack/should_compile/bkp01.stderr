[1 of 4] Processing p
unit p(hole:H)
  [1 of 2] Compiling H[sig]           ( p/H.hsig, nothing )
  [2 of 2] Compiling A                ( p/A.hs, nothing )
[2 of 4] Processing q
unit q(hole:H)
- include p(hole:H) [(A, p(hole:H):A)]
  [1 of 1] Compiling H[sig]           ( q/H.hsig, nothing )
[3 of 4] Processing h
  Instantiating h
unit h
  [1 of 1] Compiling H                ( h/H.hs, bkp01-out/h/H.o )
[4 of 4] Processing r
  Instantiating r
unit r
- include h [(H, h:H)]
- include q(h:H) []
  [1 of 2] Including h
  [2 of 2] Including q
    Instantiating q(h:H)
unit q(h:H)
- include p(hole:H) [(A, p(hole:H):A)]
    [1 of 1] Including p
      Instantiating p(h:H)
unit p(h:H)
      [1 of 2] Compiling H[sig]           ( p/H.hsig, bkp01-out/p/p-6KeuBvYi0jvLWqVbkSAZMq/H.o )
      [2 of 2] Compiling A                ( p/A.hs, bkp01-out/p/p-6KeuBvYi0jvLWqVbkSAZMq/A.o )
    [1 of 1] Compiling H[sig]           ( q/H.hsig, bkp01-out/q/q-6KeuBvYi0jvLWqVbkSAZMq/H.o )
